src/main/java/com/ust/sourcecourse/configuration/service/ColumnsService.java

```diff
@@ -0,0 +1,142 @@
+
+package com.ust.sourcecourse.configuration.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.ust.sourcecourse.configuration.entity.GroupColumn;
+import com.ust.sourcecourse.configuration.entity.ProjectGroup;
+import com.ust.sourcecourse.configuration.entity.SourceColumn;
+import com.ust.sourcecourse.configuration.repository.ProjectGroupRepository;
+import com.ust.sourcecourse.configuration.repository.SourceColumnRepository;
+import com.ust.sourcecourse.configuration.request.ColumnsRequest;
+import com.ust.sourcecourse.configuration.response.AttributesInfo;
+import com.ust.sourcecourse.configuration.response.ColumnsResponse;
+import com.ust.sourcecourse.configuration.response.SourceInfo;
+
+@Service
+public class ColumnsService {
+
+	@Autowired
+	private SourceColumnRepository columnsRepository;
+
+	@Autowired
+	private ProjectGroupRepository projectGroupRepository;
+	public List<ColumnsResponse> getColumnInfo(Long groupId) {
+		Optional<ProjectGroup> optional = projectGroupRepository.findById(groupId);
+		if (optional.isPresent()) {
+			ProjectGroup projectGroup = optional.get();
+			return getColumnResponses(projectGroup);
+
+		} else {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Project Group with uid " + groupId + " not found");
+		}
+
+	}
+
+	@Transactional
+	public List<ColumnsResponse> saveData(Long groupId, List<ColumnsRequest> columnsRequests) {
+		Optional<ProjectGroup> optional = projectGroupRepository.findById(groupId);
+
+		if (optional.isPresent()) {
+			ProjectGroup projectGroup = optional.get();
+			List<GroupColumn> groupColumns = projectGroup.getGroupColumns();
+			if (groupColumns == null) {
+				groupColumns = new ArrayList<>();
+			}
+
+			for (ColumnsRequest columnsRequest : columnsRequests) {
+				SourceColumn sourceColumn = columnsRepository.findById(columnsRequest.getSourceColumnUid())
+						.orElseThrow();
+
+				GroupColumn groupColumn = GroupColumn.builder().name(columnsRequest.getName())
+						.notes(columnsRequest.getNotes()).type(columnsRequest.getType())
+						.isPrimary(columnsRequest.isPrimary()).defaultValue(columnsRequest.getDefaultvalue())
+						.prefix(columnsRequest.getPreffix()).suffix(columnsRequest.getSuffix())
+						.projectGroup(projectGroup).sourceColumn(sourceColumn).build();
+
+				groupColumns.add(groupColumn);
+				projectGroup.setGroupColumns(groupColumns);
```

Comment:
Move line 66 & 67 after the for loop (After line 68)

Reply:
resolved